File: training/easy/ascii-art.scala
import scala.io.StdIn._

object Solution extends App {
  val l = readInt
  val h = readInt
  val t = readLine
  (0 until h)
    .map(_ => t.toCharArray.map(transformCharacter(readLine)).mkString)
    .foreach(println)

  def transformCharacter(row: String)(c: Char): String = {
    val pos = c.toUpper - 'A'
    val start = (if (pos < 0 || pos > 25) 26 else pos) * l
    row.substring(start, start + l)
  }
}

File: training/easy/ascii-art.kt
import java.util.*
import kotlin.streams.toList

fun main(args : Array<String>) {
    val input = Scanner(System.`in`)
    val l = input.nextInt()
    val h = input.nextInt()
    input.nextLine()

    val message = input.nextLine()
        .toUpperCase()
        .chars()
        .map { if (it < 'A'.toInt() || it > 'Z'.toInt()) 26 else it - 'A'.toInt() }
        .toList()
        .map { it * l to it * l + l }

    (0 until h).forEach {
        val row = input.nextLine()
        message.forEach { (start, end) -> print(row.substring(start, end)) }
        println()
    }
}
